cdap-data-fabric/src/main/java/co/cask/cdap/store/DefaultNamespaceStore.java

```diff
@@ -73,9 +73,10 @@ public NamespaceMDS get() {
             table = dsCache.getDataset(APP_META_INSTANCE_ID.getId());
           } catch (DatasetInstantiationException e) {
             try {
-              table = DatasetsUtil.getOrCreateDataset(
```

Comment:
The dataset itself (underlying table) is created only once. But the dataset object that is returned from the ``getOrCreateDataset`` is a new one every time.